### PR TITLE
CHANGES: Preliminary change notes for the 2.1.0 release

### PR DESCRIPTION
**Pull request by thisisG**
This is what I could gather of "important" things.

The major changes are generally new and/or improved APIs or changes that break backwards compatibility for one reason or another.

Please do add to or remove things from this list as you deem appropriate - this change log is far from complete.

### CODE REVIEW

File: CHANGES [hunk 1]
```diff
@@ -1,275 +1,32 @@
...
-
-
-1.0.0   Initial release
+  PR 263 - Addition of CUTest unit testing framework
+  PR 258 - Add AES-GCM to DTLS-SRTP Protection Profiles
+  PR 248 - Updated README now used to generate documentation
+  PR 239 - octet_string_set_to_zero() delegates to OPENSSL_cleanse() if
+           available, if not it will use srtp_cleanse() to zero memory
 
+  Lots of minor changes. See the pull requests for details.
```
Comment (paulej) [inline CHANGES]: Do we want to enumerate them?  I went through the list and collected the following:

#149, #151, #160, #167, #169, #173, #174, #175, #182, #191, #192, #195, #198, #202, #204, #205, #207, #208, #209, #211, #213, #214, #222, #224, #231, #232, #233, #234, #236, #237, #238, #239, #241, #243, #244, #245, #246, #248, #249, #250, #253, #255, #258, #261, #263, #265, #267, #272, #273, #274, #276, #278, #279, #283

**Comment by paulej** [comment]
Two more items that came to mind:

BoringSSL 192-bit key length support

Nice performance increase (what conditions?  @JonathanLennox  contribution -- perhaps he can remind me of the details)

**Comment by JonathanLennox** [comment]
Performance improvement is when using the OpenSSL crypto -- more or less always, I think. (The big improvement was to make AES key expansion to only be done once per key, rather than on every packet.  This helps a lot, regardless of the AES implementation you're using.)

File: CHANGES (hunk 1 above)
Comment (thisisG) [inline CHANGES]: Cool! Thanks. I will go through them and do a short tagline per PR (so basically same as Other changes, which is a fine name for the list I'd say.).

**Comment by thisisG** [comment]
@paulej, @JonathanLennox - Thanks! Both these things seems like something that will go under "Major" - will add them!

File: CHANGES [hunk 2]
```diff
@@ -1,275 +1,146 @@
...
-  Pull request 93 - Don't include AES-192 when compiling against BoringSSL.
+  PR #253 - Cipher type cleanup for AES
+    When libSRTP is compiled with OpenSSL and the AES 256 ICM is used with RTCP
+    an incorrect initialization vector is formed.
+    This change will break backwards compatibility with older versions (1.5,
+    2.0) of libSRTP when using the AES 256 ICM cipher with OpenSSL for RTCP.
 
-1.5.1
+  PR #259 - Sequence number incorrectly masked for AES GCM IV
+    The initializing vector for AES GCM encryption was incorrectly formed on
```
Comment (paulej) [inline CHANGES]: initializing => initialization

File: CHANGES [hunk 3]
```diff
@@ -1,275 +1,146 @@
...
-  Pull request 94 - Add missing copyright headers. 
+Major changes
 
-  Pull request 90 - Fix out-of-source tree builds.
+  PR #204 - OpenSSL performance improvements
+    Changed key expansion to occur once per key instead of once per packet.
 
-  Pull request 89 - Introduce little endian RISC support
+  PR #209 - Restore AES-192 under BoringSSL
+    BoringSSL supports AES-192, now enabled in libSRTP.
```
Comment (paulej) [inline CHANGES]: I would suggest replacing "," with "and is"

File: CHANGES [hunk 4]
```diff
@@ -1,275 +1,146 @@
...
 
-  Pull request 86 - Add support for cross-compiling the shared library for Windows and OS X
+  PR #224 - Master Key Identifiers (MKI) Support patch
+    Adds MKI support with up to 4 keys.
 
-  Pull request 85 - Add -f <pcap filter> option to rtp_decoder
+  PR #234 - Report SSRC instead of srtp_stream_t in srtp_event_data_t
+    srtp_stream_t is an opaque type making the event framework almost useless,
```
Comment (paulej) [inline CHANGES]: I would put a period at the end.  Start a new sentence where "Now the" is.

File: CHANGES [hunk 5]
```diff
@@ -1,275 +1,146 @@
...
+    now the SSRC is returned instead for use as a key in the public API.
 
-  Pull request 84 - Avoid problems due to unsafe macros
+  PR #238 - Configure changes and improvements
+    CFLAGS check more shell neutral, quotation fixes, always generate and
+    install pkg-config file, improved OpenSSL discovery and linking, remove
+    -fPIC flag on Windows, fix shared library generation under Cygwin, replace
+    hardcoded CFLAGS with compiler checks, regenerate configure after
```
Comment (paulej) [inline CHANGES]: Add "and" before "regenerate"?

File: CHANGES [hunk 6]
```diff
@@ -1,275 +1,146 @@
...
+    configure.in changes.
 
-  Pull request 82 - Align the AES ICM nonce
+  PR #241 & PR #261 - Improved logging API to receive log messages from libSRTP
+    Provides a logging API and the ability to enable logging to stdout and file
```
Comment (paulej) [inline CHANGES]: "and file" => "and a file".  Add a comma after "file".

File: CHANGES (hunk 2 above)
Comment (thisisG) [inline CHANGES]: Thanks @paulej! I'll get this and the others you've marked fixed.

**Pull request merged**